crypto/x509/x509_vfy.c

```diff
@@ -3372,6 +3380,36 @@ static int check_key_level(X509_STORE_CTX *ctx, X509 *cert)
     return EVP_PKEY_security_bits(pkey) >= minbits_table[level - 1];
 }
 
+/*
+ * Check whether the public key of ``cert`` does not use explicit params
+ * for an elliptic curve.
+ *
+ * Returns 1 on success, 0 otherwise.
+ */
+static int check_curve(X509 *cert)
+{
+    EVP_PKEY *pkey = X509_get0_pubkey(cert);
+
+    /* Unsupported or malformed keys are not secure */
+    if (pkey == NULL)
+        return 0;
+
+#ifndef OPENSSL_NO_EC
+    if (EVP_PKEY_id(pkey) == EVP_PKEY_EC) {
+        const EC_GROUP *grp;
+
+        grp = EC_KEY_get0_group(EVP_PKEY_get0_EC_KEY(pkey));
+        if (!grp)
+            return 0;
+
+        if (!(EC_GROUP_get_asn1_flag(grp) & OPENSSL_EC_NAMED_CURVE))
```

Comment:
You cant do this as the test - this is just an option that says 'when we serialize the ec key save it in named curve format" i.e: A named curve can have this option.
Use EC_GROUP_get_curve_name instead.

Reply:
Hmm but it at least somehow works now. However `EC_GROUP_get_curve_name(grp) == NID_undef` actually does not, because if the parameters match a known curve the NID will be set accordingly. To be compliant with the RFC we should reject all certs with explicit parameters even if they happen to match an existing named curve.